python/tests/integration/arcticdb/test_arctic.py

```diff
@@ -293,41 +293,125 @@ def test_parallel_writes_and_appends_index_validation(arctic_library, finalize_m
 
 @pytest.mark.parametrize("finalize_method", (StagedDataFinalizeMethod.APPEND, StagedDataFinalizeMethod.WRITE))
 class TestFinalizeWithEmptySegments:
-    def test_staged_segment_is_only_empty_dfs(self, lmdb_library, finalize_method):
-        lib = lmdb_library
+    def test_staged_segment_is_only_empty_dfs(self, arctic_library, finalize_method):
+        lib = arctic_library
         lib.write("sym", pd.DataFrame([]), staged=True)
         lib.write("sym", pd.DataFrame([]), staged=True)
         lib.finalize_staged_data("sym", mode=finalize_method)
         assert_frame_equal(lib.read("sym").data, pd.DataFrame([], index=pd.DatetimeIndex([])))
 
-    def test_staged_segment_has_empty_df(self, lmdb_library, finalize_method):
-        lib = lmdb_library
+    def test_staged_segment_has_empty_df(self, arctic_library, finalize_method):
+        lib = arctic_library
         index = pd.DatetimeIndex([pd.Timestamp(2024, 1, 1), pd.Timestamp(2024, 1, 3), pd.Timestamp(2024, 1, 4)])
         df1 = pd.DataFrame({"col": [1, 2, 3]}, index=index)
         df2 = pd.DataFrame({})
         df3 = pd.DataFrame({"col": [4]}, index=pd.DatetimeIndex([pd.Timestamp(2024, 1, 5)]))
         lib.write("sym", df1, staged=True)
         lib.write("sym", df2, staged=True)
         lib.write("sym", df3, staged=True)
-        lib.finalize_staged_data("sym", mode=finalize_method)
-        assert_frame_equal(lib.read("sym").data, pd.concat([df1, df2, df3]))
+        with pytest.raises(SchemaException):
+            lib.finalize_staged_data("sym", mode=finalize_method)
 
-    def test_df_without_rows(self, lmdb_library, finalize_method):
-        lib = lmdb_library
+    def test_df_without_rows(self, arctic_library, finalize_method):
+        lib = arctic_library
         df = pd.DataFrame({"col": []}, index=pd.DatetimeIndex([]))
         lib.write("sym", df, staged=True)
         lib.finalize_staged_data("sym", mode=finalize_method)
         assert_frame_equal(lib.read("sym").data, df)
 
 @pytest.mark.parametrize("finalize_method", (StagedDataFinalizeMethod.APPEND, StagedDataFinalizeMethod.WRITE))
-def test_finalize_without_adding_segments(lmdb_library, finalize_method):
-    lib = lmdb_library
+def test_finalize_without_adding_segments(arctic_library, finalize_method):
+    lib = arctic_library
     with pytest.raises(UserInputException) as exception_info:
         lib.finalize_staged_data("sym", mode=finalize_method)
 
+class TestFinalizeStagedDataSchemaMismatch:
```

Comment:
`finalize_staged_data` just passes arguments through to the V1 API, so the vast majority of the tests should be on the V1 API. So either move this to `test_parallel.py`, or remove it if it is duplicating tests already covered there

Reply:
Ah, I wasn't aware of `test_parallel.py` (when it comes to finalize.* methods the file naming is somewhat random). I thought that this is the place where `finalize_staged_data` is tested. I thing the correct place for this is `test_parallel.py`, moving it there.